nltk/tokenize/punkt.py

```diff
@@ -1396,12 +1396,13 @@ def _match_potential_end_contexts(self, text: str) -> Iterator[Tuple[Match, str]
 
             # Get the slice of the previous word
             before_text = text[previous_slice.stop : match.start()]
-            last_space_index = self._get_last_whitespace_index(before_text)
-            if last_space_index:
-                last_space_index += previous_slice.stop
+            index_after_last_space = self._get_last_whitespace_index(before_text)
+            if index_after_last_space:
+                # + 1 to exclude the space itself
+                index_after_last_space += previous_slice.stop + 1
             else:
-                last_space_index = previous_slice.start
-            prev_word_slice = slice(last_space_index, match.start())
+                index_after_last_space = previous_slice.start
+            prev_word_slice = slice(index_after_last_space, match.start())
```

Comment:
This method, which is supposed to return a "context" word that comes before a specific token of interest, used to also return the whitespace that preceded that word. I've fixed this now, so it doesn't return the whitespace anymore. It should not affect the tokenization, as that already `strip()`-ed away the extra whitespace anyways.